python/ray/ml/predictor.py

```diff
@@ -1,25 +1,60 @@
 import abc
-from typing import Union, TYPE_CHECKING
+from typing import Dict, Type, Union
 
 from ray.ml.checkpoint import Checkpoint
+from ray.util.annotations import DeveloperAPI, PublicAPI
 
-if TYPE_CHECKING:
-    import numpy as np
-    import pandas as pd
+import numpy as np
+import pandas as pd
 
-DataBatchType = Union["pd.DataFrame", "np.ndarray"]
+DataBatchType = Union[np.ndarray, pd.DataFrame, Dict[str, np.ndarray]]
```

Comment:
Adding Dict[str, np.ndarray] as a supported type.

Reply:
We should also have an Arrow Table as a `DataBatchType` option, right?